### PR TITLE
fix: improved the conditions for determining voucher subtypes

### LINKED ISSUE #39382: Incorrect Voucher Subtype
### Information about bug

As a result of commit [e5d73c7](https://github.com/frappe/erpnext/commit/e5d73c780bc5e272dd7c3db7db65665010ef6623), voucher subtype is being incorrectly returned. For e.g. When voucher is "Sales invoice" voucher subtype is being returned as "Debit Note" even when it isn't

### Module

accounts

### Version

ERPNext: v15.10.0 (version-15)
Frappe Framework: v15.9.0 (version-15)
Frappe HR: v15.9.0 (version-15)
India Compliance: v15.4.1 (version-15)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

_No response_

### PR DESCRIPTION
**Pull request by ljain112**
- Removed Debit Note And Credit Note for Sales Invoice and Purchase Invoice because it was confusing for users.
- Added Debit Note when Sales Invoice and `is_debit_note`

Closes: https://github.com/frappe/erpnext/issues/39382, https://github.com/frappe/erpnext/issues/39382
Frappe Support Issue: https://support.frappe.io/app/hd-ticket/18562

backport version-15-hotfix
backport version-14-hotfix

### CODE REVIEW

**Comment by vorasmit** [comment]
@ruthra-kumar, Should we patch this?

**Comment by ruthra-kumar** [comment]
> @ruthra-kumar, Should we patch this?

Yes.

When this field was introduced, we intentionally didn't patch old data. So, while patching now, we only have to worry about rows with `voucher_subtype` already set.

File: erpnext/patches/v15_0/update_sub_voucher_type_in_gl_entries.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+def execute():
+	update_purchase_invoices()
+	update_sales_invoices()
+	update_sales_debit_notes()
+
+
+def update_purchase_invoices():
+	invoices = frappe.get_all(
+		"Purchase Invoice",
+		filters={"docstatus": 1, "is_return": 0},
```
Comment (vorasmit) [inline erpnext/patches/v15_0/update_sub_voucher_type_in_gl_entries.py:13]: Thought:

1. Number of invoices with returns would be less, We can update subtype where it's not in this list. 
2. docstatus check can be ignored ?? we have gls for cancelled documents too.

**Comment by ruthra-kumar** [comment]
@ljain112 

Is there any blocker for this change?

**Review by vorasmit — APPROVED** [review]

**Comment by vorasmit** [comment]
LGTM.
@ruthra-kumar, you may review if you want, and merge this.

**Review by ruthra-kumar — APPROVED** [review]

**Pull request merged**